applications/fishing-map/src/features/map/Map.tsx

```diff
@@ -211,18 +199,36 @@ const MapWrapper = (): React.ReactElement | null => {
         </InteractiveMap>
       )}
       <MapControls loading={tilesLoading.loading} />
-      {layersWithLegend?.map(
-        (legend) =>
-          document.getElementById(legend.id as string) &&
-          createPortal(
+      {layersWithLegend?.map((legend) => {
+        const legendDomElement = document.getElementById(legend.id as string)
+        if (legendDomElement) {
+          const isSquareKm = (legend.gridArea as number) > 50000
+          const gridArea = isSquareKm ? (legend.gridArea as number) / 1000000 : legend.gridArea
+          const gridAreaFormatted = gridArea
+            ? formatI18nNumber(gridArea, {
```

Comment:
Any reason for not putting that logic inside `getLegendLayers` ? This seems pretty generic

Reply:
You are right, the parsing function should be there and in the component just the custom styles.